### PR TITLE
Make files execution order predictive

### PR DESCRIPTION
**Pull request by duailibe**
Ever since I started contributing to prettier, the `tests_integration` tests fail for me because of file ordering when there's a uppercase filename (e.g. `cli/jest/__best-tests__` and `cli/jest/Component.js`). Also recently a snapshot were rewrited for the same reason (see: https://github.com/prettier/prettier/pull/3187#issuecomment-342487311 and #3189).

Since this effectively affects the CLI (not drastically I hope), we could simply rename the test files, but that's really a temporary fix. Otherwise, I'm not sure how else to prevent this issue.

### CODE REVIEW

File: src/cli-util.js [hunk 1]
```diff
@@ -271,6 +271,8 @@ function eachFilename(argv, patterns, callback) {
       .sync(patterns, { dot: true })
       .map(filePath => path.relative(process.cwd(), filePath));
 
+    filePaths.sort((a, b) => a.toLowerCase().localeCompare(b.toLowerCase()));
```
Comment (lydell) [inline src/cli-util.js]: Just a curious question: Is there a specific reason you use `.localeCompare` instead of just `filePaths.sort()`?

Comment (duailibe) [reply src/cli-util.js]: @lydell `filePaths.sort()` sorts using case sensitivity (i.e. the same as `filePaths.sort((a, b) => a.localeCompare(b))`

Comment (lydell) [reply src/cli-util.js]: Sorry, let me rephrase my question: Is there any specific reason you don’t just use `filePaths.sort()`? Nicer output? Or is it really required to do it case insensitively?

Comment (duailibe) [reply src/cli-util.js]: The output of `glob` is already sorted. But the case sensitivity is the same for matching (if you want case sensitive matches, it will sort with case sensitivity).

https://github.com/isaacs/node-glob/blob/master/common.js#L153

**Comment by duailibe** [comment]
That's fair I'm not that happy with this either
